renderite/serialization.zig

```diff
@@ -65,6 +76,14 @@ pub const IpcDeserializer = struct {
 
         return try self.reader.readSliceEndianAlloc(gpa, u16, @intCast(len), endian);
     }
+
+    pub fn readValueList(self: IpcDeserializer, comptime T: type, gpa: std.mem.Allocator) ![]T {
+        const len = try self.reader.takeInt(i32, endian);
+        if (len == 0 or len == -1)
+            return &.{};
```

Comment:
If values *less* than `-1` are explicitly invalid, then instead of this suggestion, `< -1` should be checked, and an error should be thrown in that case.

```suggestion
        if (len <= 0)
            return &.{};
```

Reply:
-1 actually means `null`, so I need to implement logic for that. This should be a TODO:

```suggestion
        if (len == 0 or len == -1) // TODO: handle -1 meaning null
            return &.{};
```